tests/integration-service/integration.go

```diff
@@ -123,161 +118,236 @@ var _ = framework.IntegrationServiceSuiteDescribe("Integration Service E2E tests
+				Expect(err).NotTo(HaveOccurred())
+
+				createApp()
+				createComponent()
+
+				_, err = f.AsKubeAdmin.IntegrationController.CreateIntegrationTestScenario(applicationName, appStudioE2EApplicationsNamespace, BundleURLFail, InPipelineNameFail)
+				Expect(err).ShouldNot(HaveOccurred())
 			})
 
-			It("checks if a Release is created successfully", func() {
-				timeout = time.Second * 800
-				interval = time.Second * 10
-				Eventually(func() bool {
-					if f.AsKubeAdmin.IntegrationController.HaveHACBSTestsSucceeded(applicationSnapshot_push) {
-						releases, err := f.AsKubeAdmin.IntegrationController.GetReleasesWithApplicationSnapshot(applicationSnapshot_push, appStudioE2EApplicationsNamespace)
-						Expect(err).ShouldNot(HaveOccurred())
-						if len(*releases) != 0 {
-							for _, release := range *releases {
-								GinkgoWriter.Printf("Release %s is found\n", release.Name)
-							}
-						} else {
-							Fail("No Release found")
-						}
-						return true
-					}
-					applicationSnapshot_push, err = f.AsKubeAdmin.IntegrationController.GetApplicationSnapshot(applicationSnapshot_push.Name, "", appStudioE2EApplicationsNamespace, "")
-					if err != nil {
-						GinkgoWriter.Printf("snapshot %s has not been found yet\n", applicationSnapshot_push.Name)
-					}
-					return false
-				}, timeout, interval).Should(BeTrue(), "time out when waiting for release created")
+			AfterAll(func() {
+				if !CurrentSpecReport().Failed() {
+					cleanup()
+				}
 			})
 
-			It("checks if an EnvironmentBinding is created successfully", func() {
-				timeout = time.Second * 600
-				interval = time.Second * 2
-				Eventually(func() bool {
-					if f.AsKubeAdmin.IntegrationController.HaveHACBSTestsSucceeded(applicationSnapshot_push) {
-						envbinding, err := f.AsKubeAdmin.IntegrationController.GetSnapshotEnvironmentBinding(applicationName, appStudioE2EApplicationsNamespace, env)
-						Expect(err).ShouldNot(HaveOccurred())
-						Expect(envbinding != nil).To(BeTrue())
-						GinkgoWriter.Printf("The EnvironmentBinding is created\n")
-						return true
-					}
-					applicationSnapshot_push, err = f.AsKubeAdmin.IntegrationController.GetApplicationSnapshot(applicationSnapshot_push.Name, "", appStudioE2EApplicationsNamespace, "")
-					if err != nil {
-						GinkgoWriter.Printf("snapshot %s has not been found yet\n", applicationSnapshot_push.Name)
-					}
-					return false
-				}, timeout, interval).Should(BeTrue(), "time out when waiting for release created")
+			It("triggers a build PipelineRun", Label("integration-service"), func() {
+				assertBuildPipelineRunFinished()
+			})
+
+			It("checks if the ApplicationSnapshot is created", func() {
+				assertApplicationSnapshotCreated()
+			})
+
+			It("checks if all of the integrationPipelineRuns finished", Label("slow"), func() {
+				assertIntegrationPipelineRunFinished()
+			})
+
+			It("checks if snapshot is marked as failed", func() {
```

Comment:
Do we also want to double check that no `SnapshotEnvironmentBinding` or `Release` CRs are created for this failed snapshot?

Reply:
https://github.com/redhat-appstudio/e2e-tests/blob/main/tests/integration-service/integration.go#L160

I could, but I'm somehow confused by this code, too many things are happening there

So I should repeat everything: creating release plan, env, push event, check tests again and then check if GCL is NOT updated and release was NOT created?

Reply:
You have a point there. I think that it's enough to verify that the GCL is not updated - since that doesn't require recreating the ReleasePlan or Environment.

Reply:
The current solution looks good to me, thanks for following up